components/admin_console/workspace-optimization/dashboard.tsx

```diff
@@ -120,17 +113,34 @@ const WorkspaceOptimizationDashboard = (props: Props) => {
             }
 
             setVersionData({type, version: result.tag_name, status});
-            return;
+        }
+    };
+
+    const fetchGuestAccounts = async () => {
+        if (TeamSettings?.EnableOpenServer && GuestAccountsSettings?.Enable) {
+            let usersArray = await fetch('/api/v4/users/invalid_emails').then((result) => result.json());
```

Comment:
Should we pass some query parameters for page & per_page? Just in case it turns out there are thousands of records returned?

Reply:
good question ... is this a very common use case? And how do we determine if we need to fetch all?

Reply:
I don't ever think we should fetch all, just fetch the first 50 here

Reply:
Yeah, I know what you mean, but consider this: For a server that has 20K+ users having 5 guests is pretty reasonable and would not be very actionable. On the other hand a server with 50 users and 5 guest accounts this is a different story

Reply:
I don't mind keeping the threshold at 5% if we can at least limit the number of users returned from that endpoint

Reply:
well ... maybe we can adjust the API a bit to accept a parameter to only return the number of users with email domains that differ from the server domain? On the other hand that would mean we would not be able to check for whitelisted domains

Reply:
So a total count of invalid users? That's a good idea but with our current timelines I don't think we have time for any more adjustments.

@mkraft what's your take on this?

Reply:
I think if it's not clearly defined and performant by this point this widget can wait for a future iteration. It was a nice-to-have anyhow and we don't want to delay the rest of our great work because of it.

Reply:
ok, so for now we can just merge it as is and revisit later, right?

Any objections @BenCookie95 ?

Reply:
My intents are:
* avoid delaying getting this to QA
* no big performance slowdown if it makes it in

Reply:
question is: is the API slowing down significantly if the numbers are high? Can we test that somehow, maybe on community-daily?

Reply:
OK, I will remove that metric for now and we can re-visit when the MVP is out